src/v/cloud_roles/refresh_credentials.cc

```diff
@@ -225,6 +225,13 @@ refresh_credentials::impl::calculate_sleep_duration(uint32_t expiry_sec) const {
     int sleep = std::floor(expiry_sec * sleep_from_expiry_multiplier);
     vlog(
       clrl_log.trace, "sleep duration adjusted with buffer: {} seconds", sleep);
+    if (sleep < 0) {
```

Comment:
This happens in unit tests sometimes, if the payload is static (ie contains a fixed datetime), the diff can end up being negative (I saw this several times when running the `test_aws_credentials` unit test. 

The major side effect is that the test hangs with the sleep interval being negative.

Reply:
This function makes my brain hurt a little. The input is uint32_t, then we do a float conversion, then we do an int conversion and only here we check for negative result. If we remove the narrowing conversion to int, then we won't get any negative results and we won't need this hack.

It would still be possible to get absurdly high sleep values. But, I believe, I just showed that if we remove the bug from this function we uncovered a possible bug in the caller who calls this method with negative durations. It is the one who needs to be fixed (after fixing the bug here introduced by narrowing conversion).

You still might say that a sanity check is useful. I agree, but that sanity check must be introduced where the value is actually used using the relevant context. In this case we have the `co_await ss::sleep_abortable(*_sleep_duration, _as);` call. Negative sleep durations here don't make sense. You can also do a higher cap.

As an alternative, if you still prefer the current logic, at least make expiry_sec a chrono::duration type and clamp it to a non-negative value.